src/reify.ml4

```diff
@@ -11,9 +11,29 @@ let _ = Goptions.declare_bool_option {
   Goptions.optread = (fun () -> !cast_prop);
   Goptions.optwrite = (fun a -> cast_prop:=a);
 }
-
+                                     
 let is_cast_prop () = !cast_prop                     
-                     
+
+let share_strings = ref (false)
+let _ = Goptions.declare_bool_option {
+  Goptions.optsync = true; Goptions.optdepr = false;
+  Goptions.optname = "Maximal sharing of ascii/strings";
+  Goptions.optkey = ["Template";"Share";"Strings"];
+  Goptions.optread = (fun () -> !share_strings);
+  Goptions.optwrite = (fun a -> share_strings:=a);
+}
```

Comment:
Maybe expose the option to let-bind strings, since that would change something like [quote (quote a)].